goon/code/datums/browserOutput.dm

```diff
@@ -198,16 +198,26 @@ var/list/chatResources = list(
 
 	return "<img [class] src='data:image/png;base64,[bicon_cache[key]]'>"
 
+/proc/is_valid_tochat_message(message)
+	return istext(message)
+
+/proc/is_valid_tochat_target(target)
+	return !istype(target, /savefile) && (ismob(target) || islist(target) || isclient(target) || target == world)
+
 var/to_chat_filename
 var/to_chat_line
 var/to_chat_src
 // Call using macro: to_chat(target, message)
 /proc/__to_chat(target, message)
-	if(istype(message, /image) || istype(message, /sound) || istype(target, /savefile) || !(ismob(target) || islist(target) || isclient(target) || target == world))
+	if(!is_valid_tochat_message(message) || !is_valid_tochat_target(target))
 		target << message
 		if(!istext(message))
 			message = "(non-text type)"
-		world.Error(new/exception("DEBUG: to_chat called with invalid message: [message]", to_chat_filename, to_chat_line), e_src = to_chat_src)
+		var/targetstring = "\'[target]\'"
```

Comment:
Unnecessary quote escaping. Also, the line above this one - would probably be handy if a null message was differentiated from a non-null, non-text message. Same for null targets. 

I bet there's quite a few things that mishandle nulls. Still not sure if we should just ignore them, or continue treating them as errors.